### DISCUSSION TITLE
how to build this project?

### DISCUSSION THREAD

**Discussion by bestsung** [FrozenFish259/Goose_Goose_Duck_Hack#166] [Q&A 疑难解答区] [answered]
how to build this project?
i have warning and error T^T
thanks for reply
(now not support thai language)
![gg](https://user-images.githubusercontent.com/23060428/213860355-0ce0ea43-07d6-4bef-897e-4597494483bd.png)

**Comment by FrozenFish259** [comment, marked as answer]
Double click to open .sln file. Build with Release x64

**Reply by bestsung** [reply]
how can i change font for support other language

**Reply by FrozenFish259** [reply]
UI.cpp:375 Edit the path to font.
UI.cpp:375 Change the glyph_range to `ImGui::GetIO().Fonts->GetGlyphRangesThai());`

**Reply by bestsung** [reply]
![image](https://user-images.githubusercontent.com/23060428/213862979-71f90f73-7d02-49d4-856f-44dcc07789c2.png)
noo not working thai language is ????

**Reply by FrozenFish259** [reply]
> ![image](https://user-images.githubusercontent.com/23060428/213862979-71f90f73-7d02-49d4-856f-44dcc07789c2.png) noo not working thai language is ????

That's weird. Those `???` are English text which should be shown without any problems. However, I am not familiar with ImGui

**Reply by bestsung** [reply]
English is no problem. I use Thai language. It becomes `????`

**Reply by bestsung** [reply]
![image](https://user-images.githubusercontent.com/23060428/213863481-eef6b4d1-0697-4565-997f-dbbce93e30b1.png)

**Reply by FrozenFish259** [reply]
> English is no problem. I use Thai language. It becomes `????`

Maybe there are no Thai characters in your ttf font

**Reply by bestsung** [reply]
my tff is thai font

**Reply by FrozenFish259** [reply]
> my tff is thai font

I think the problem should be Drawing.cpp:26

**Reply by FrozenFish259** [reply]
Change`#define str(eng,cn) utils.b_chineseOS?(const char*)u8##cn:eng`
to `#define str(eng,cn) utils.b_chineseOS?(const char*)u8##cn:(const char*)u8##eng`

**Reply by FrozenFish259** [reply]
Because I was not using u8 string on English os
